### ISSUE TITLE
'Send to interactive console': interrupt parsing when there is an error

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#115]
_From [david.fo...@gmail.com](https://code.google.com/u/107726689308966932341/) on 2009-11-10T17:11:49Z_

I mentionned the fact that interactive console was sometime very slow to
evaluate the lines of my code, which is annoying for reevaluation of a full
class, etc... I did not open an issue at that time, but now I found out a
big part of the reason by looking at the issue on slow big variables.

I use some very big arrays in my work, and in fact, when I start to
accumulate some big variables, the console becomes slow, and it is solved
when I disable auto-refresh. It then gets pretty much as fast as external
console parsing

I can also suggest to interrupt parsing when there is an error, so for a
big selection, it is not necessary to wait for a long list of red errors
because of some indent error in the beginning. That is good for internal
and external console. What is the expected output? What do you see instead? What version of the product are you using? On what operating system? spyder 1.0.0rc2 on kubuntu karmic 

Please provide any additional information below

.

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=115_

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2010-01-03T11:27:49Z_

**Summary:** 'Send to interactive console': interrupt parsing when there is an error  
  **Status:** Accepted

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2010-01-06T02:43:07Z_

After some testing and bugfixing around this 'Send to interactive console' feature, I 
found out that the fact that it's not interrupting parsing when there is an error is 
the default Python interpreter behavior.

Indeed, in a standard Python interpreter (simply run python.exe) if you copy/paste a 
group of lines with an indentation error, Python won't stop at the first error.

That's why I'm closing this issue even if the original idea was a good one (I tried 
for some time to make it work without success... and then, when I found out that it 
was the interpreter default behavior, I understood why this bug was so tenacious).

**Status:** WontFix
